### DISCUSSION TITLE
Rename "ollama" to "Custom"

### DISCUSSION THREAD

**Discussion by gnusupport** [Taewan-P/gpt_mobile#148] [General]
First I appreciate you made this application and it is free software under GNU GPL license. 

I am using it heavily with the locally running models, each of those models is fully free software such as Qwen, Rocket, Phi-3.5, Mistral, and I am using exclusively those versions which are free software as for better future sharing. 

Now Llama is not free software, their origin is Meta and they have quite different purposes with it. 

The word Llama is now inside of various LLM running software programs like ollama or llamafile or llama.cpp

Though none of them requires Llama from Meta to run it.

I don't think that Llama deserves attention as their are liars and sneaking into free software community by lying they got "open source" while not following the true open source definition. Their software is restricted.

And in this state software there is option for ollama, but then I never installed ollama and I am using that option to run models running with llamafile or llama.cpp so it's really not correct.

And who knows which other methods are there to run it locally like vllm or airllm.

That's why I suggest that option Ollama is renamed to Custom or something else similar not related to ollama.

And such option should be basically first in the menu, IMHO to drive people to install it locally, and to appreciate and foster the truly free software.

I would even rename GPTMobile to something like "LLM Chat". As not each language model is GPT, but GPT is also LLM.

As default under Groq, I would put Mixtral option as it is free software.

**Comment by Taewan-P** [comment]
Hi, Thanks for the information.
I am currently preparing for huge refactor & redesign for many of the feature requests that I received during the past few months. I will be considering this in the rework too. Thanks

**Reply by AbhiAbzs** [reply]
Thank you so much for creating this app.
I was also thinking of requesting a feature similar to this, that apart from Ollama, can you please also option to integrate other options such as LMStudio, Jan or Cortex Ai, vLLM, options like Openrouterr etc.

There is already a closed source option in playstore which you can check for inspiration, called **Crosstalk** Multi-LLM AI Chat by Ham and More.

**Comment by Taewan-P** [comment]
Hello, this feature is now implemented in [v0.7.0](https://github.com/Taewan-P/gpt_mobile/releases). Thanks!
cc @AbhiAbzs

**Reply by AbhiAbzs** [reply]
Thank you so much 🙏
I'll test this ASAP

**Reply by AbhiAbzs** [reply]
It seems like the build hasn't been updated on playstore or fdroid for past 1 to 2 years.

**Comment by gnusupport** [comment]
Thanks, your app is simple and works great, even though I would like to make fork.
